process-manager.js

```diff
@@ -42,47 +62,66 @@ class ProcessWrapper extends EventEmitter {
 		this.process.disconnect();
 	}
 
+	/**
+	 * @return {number}
+	 */
 	get load() {
 		return this.pendingTasks.size;
 	}
 }
 
-// execFile - the path to the file to spawn the child process(es) from
-// maxProcesses - the maximum number of child processes to spawn
-// isChatBased - the process manager handles some chat functionality
+/**
+ * @typedef {Object} PMOptions
+ * The path to the file to spawn the child process(es) from.
+ * @property {string} [execFile]
+ * The maximum number of child processes to spawn.
+ * @property {number} [maxProcesses]
+ * Whether or not the process manager handles a chat commands module.
+ * @property {boolean} [isChatBased]
+ */
+
+/**
+ * @class ProcessManager
+ * @property {Set<ProcessWrapper>} processes
+ * @property {number} taskId
+ * @property {string} execFile
+ * @property {number} maxProcesses
+ * @property {boolean} isChatBased
+ */
 class ProcessManager {
+	/**
+	 * @param {PMOptions} options
+	 */
 	constructor(options) {
-		if (!('execFile' in options)) {
-			Monitor.debug(
-				"No execFile property was missing form the options object to be " +
-				"given to the ProcessManager constructor!"
-			);
-		} else if (!('maxProcesses' in options) || !('isChatBased' in options)) {
-			Monitor.debug(
-				"An options object given to the ProcessManager constructor is " +
-				"missing required properties! The filename given is: " + (options.execFile || '""') + "."
+		if (!('execFile' in options) || !('maxProcesses' in options) || !('isChatBased' in options)) {
+			throw new Error(
+				"An options object given to the ProcessManager constructor is missing required properties! " +
+				`The execFile given is: ${options.execFile || ''}`
 			);
 		}
 
-		this.processes = [];
+		this.processes = new Set();
 		this.taskId = 0;
-		this.execFile = options.execFile;
-		this.maxProcesses = (typeof options.maxProcesses === 'number') ? options.maxProcesses : 1;
+		this.execFile = '' + options.execFile;
+		this.maxProcesses = +options.maxProcesses || 0;
```

Comment:
We have been here, *too*... Same thread :|
No reason to change this line at all.